### PR TITLE
Drop deprecated ioutil

### PR DESCRIPTION
**Pull request by kke**
Replaces ioutil usages with the modern alternatives.

Includes some gofmt created import reordering on the modified files.

### CODE REVIEW

**Review by jnummelin — APPROVED** [review]

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -13,14 +12,14 @@ import (
 
 	"github.com/docker/docker/api/types"
 	"github.com/ghodss/yaml"
-	"github.com/mitchellh/go-homedir"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
 	"github.com/k0sproject/footloose/pkg/config"
+	"github.com/k0sprojecs/footloose/pkg/config"
```
Comment (kke) [inline pkg/cluster/cluster.go:16]: Double import

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -13,14 +12,14 @@ import (
...
 	"github.com/k0sproject/footloose/pkg/docker"
 	"github.com/k0sproject/footloose/pkg/exec"
 	"github.com/k0sproject/footloose/pkg/ignite"
-)
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus
```
Comment (kke) [inline pkg/cluster/cluster.go:22]: Missing `)`

File: pkg/cluster/cluster.go [hunk 3]
```diff
@@ -598,11 +597,15 @@ func (f *matchFilter) Write(p []byte) (n int, err error) {
 }
 
 // Matches:
-//   ssh_exchange_identification: read: Connection reset by peer
+//
+//
+//	h_exchange_identification: read: Connection reset by peer
```
Comment (kke) [inline pkg/cluster/cluster.go:602]: Unrelated

File: pkg/cluster/cluster.go [hunk 4]
```diff
@@ -598,11 +597,15 @@ func (f *matchFilter) Write(p []byte) (n int, err error) {
...
 var connectRefused = regexp.MustCompile("^ssh_exchange_identification: ")
 
 // Matches:
-//   Warning:Permanently added '172.17.0.2' (ECDSA) to the list of known hosts
+//
+//
```
Comment (kke) [inline pkg/cluster/cluster.go:607]: Unintended change
